src/main/scala/algolia/definitions/DeleteDefinition.scala

```diff
@@ -44,7 +44,9 @@ case class DeleteObjectDefinition(
   def index(ind: String): DeleteObjectDefinition = copy(index = Some(ind))
 
   def objectId(objectId: String): DeleteObjectDefinition =
-    copy(oid = Some(objectId))
+    if (objectId.trim.isEmpty)
+      throw new IllegalArgumentException("objectID can't be empty or whitespaces")
```

Comment:
Oops! sorry first time coding scala..
Where should I put this validity check so ? 
In the build() method ? 
Thanks!

Reply:
I've discussed quickly with @ElPicador but he's currently at a conference. He suggested one thing that would be to introduce a dedicated type to pass as a single parameter of the definition so that both the index and the objectID can be made mandatory and non-empty.

I can help with that @Ant-hem, let me know when you'll have time.